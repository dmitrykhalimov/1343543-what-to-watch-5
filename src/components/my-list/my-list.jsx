import React, {useEffect} from "react";
import PropTypes from "prop-types";
import Logo from "../logo/logo";
import FilmsCatalog from "../films-catalog/films-catalog";
import FilmsList from "../films-list/films-list";
import Footer from "../footer/footer";
import {validFilm} from "../../utils/props";
import {connect} from "react-redux";
import {getFavorite} from "../../store/reducers/selectors";
import UserBlock from "../user-block/user-block";
import {fetchFavorites} from "../../store/api-actions";

const MyList = (props) => {
  const {films, onPageLoad} = props;

  useEffect(() => {
    onPageLoad();
  }, []);

  return (
    <div className="user-page">
      <header className="page-header user-page__head">
        <Logo/>

        <h1 className="page-title user-page__title">My list</h1>

        <UserBlock/>
      </header>

      <FilmsCatalog>
        <FilmsList
          films = {films}
          maxQuantity = {films.length}
        />
      </FilmsCatalog>

      <Footer
        isLight={true}
      />
    </div>
  );
};

MyList.propTypes = {
  films: PropTypes.arrayOf(validFilm).isRequired,
  onPageLoad: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  films: getFavorite(state),
});

const mapDispatchToProps = (dispatch) => ({
  onPageLoad() {
    dispatch(fetchFavorites());
  },
});

export {MyList};
export default connect(mapStateToProps, mapDispatchToProps)(MyList);


