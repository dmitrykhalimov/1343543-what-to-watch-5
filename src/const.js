export const ratingRanks = [
  {
    'Bad': 2.9
  },
  {
    'Normal': 4.9
  },
  {
    'Good': 7.9
  },
  {
    'Very good': 9.9
  },
  {
    'Awesome': 10
  }
];
