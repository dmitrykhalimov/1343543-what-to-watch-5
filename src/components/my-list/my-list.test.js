import React from "react";
import renderer from "react-test-renderer";
import {MyList} from "./my-list";
import {TEST_MOCKS, TEST_MOCK_STORE} from "../../const";
import {MemoryRouter} from "react-router-dom";
import configureMockStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureMockStore();
const store = mockStore(TEST_MOCK_STORE);

// TeamPage.test.js

describe(`MyList`, () => {
  it(`Should MyList render correctly`, () => {
    const tree = renderer
    .create(
        <Provider store={store}>
          <MemoryRouter>
            <MyList
              films={TEST_MOCKS.films}
              handlePageLoad={TEST_MOCKS.noop}
            />,
          </MemoryRouter>
        </Provider>
    )
  .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
