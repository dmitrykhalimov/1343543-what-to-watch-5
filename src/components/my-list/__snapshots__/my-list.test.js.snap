// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList Should MyList render correctly 1`] = `
Array [
  <div
    className="user-page"
  >
    <header
      className="page-header user-page__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <h1
        className="page-title user-page__title"
      >
        My list
      </h1>
      <div
        className="user-block"
      >
        <a
          href="/mylist"
          onClick={[Function]}
        >
          <div
            className="user-block__avatar"
          >
            <img
              alt="User avatar"
              height="63"
              src="https://i.kym-cdn.com/photos/images/newsfeed/000/607/429/84b.jpg"
              width="63"
            />
          </div>
        </a>
      </div>
    </header>
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            className="small-movie-card__link"
            href="/films/29"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <video
                autoPlay={false}
                height="175"
                muted={true}
                poster="https://thenewswheel.com/wp-content/uploads/2014/08/Interstate-60-review-BMW-convertible-James-Marsden-8-ball.jpg"
                src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
                width="280"
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              <span
                className="small-movie-card__link"
              >
                Interstate 60: Episodes of the Road
              </span>
            </h3>
          </a>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            className="small-movie-card__link"
            href="/films/29"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <video
                autoPlay={false}
                height="175"
                muted={true}
                poster="https://thenewswheel.com/wp-content/uploads/2014/08/Interstate-60-review-BMW-convertible-James-Marsden-8-ball.jpg"
                src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
                width="280"
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              <span
                className="small-movie-card__link"
              >
                Interstate 60: Episodes of the Road
              </span>
            </h3>
          </a>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            className="small-movie-card__link"
            href="/films/29"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <video
                autoPlay={false}
                height="175"
                muted={true}
                poster="https://thenewswheel.com/wp-content/uploads/2014/08/Interstate-60-review-BMW-convertible-James-Marsden-8-ball.jpg"
                src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
                width="280"
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              <span
                className="small-movie-card__link"
              >
                Interstate 60: Episodes of the Road
              </span>
            </h3>
          </a>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            className="small-movie-card__link"
            href="/films/29"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <video
                autoPlay={false}
                height="175"
                muted={true}
                poster="https://thenewswheel.com/wp-content/uploads/2014/08/Interstate-60-review-BMW-convertible-James-Marsden-8-ball.jpg"
                src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
                width="280"
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              <span
                className="small-movie-card__link"
              >
                Interstate 60: Episodes of the Road
              </span>
            </h3>
          </a>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <a
            className="small-movie-card__link"
            href="/films/29"
            onClick={[Function]}
          >
            <div
              className="small-movie-card__image"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <video
                autoPlay={false}
                height="175"
                muted={true}
                poster="https://thenewswheel.com/wp-content/uploads/2014/08/Interstate-60-review-BMW-convertible-James-Marsden-8-ball.jpg"
                src="https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm"
                width="280"
              />
            </div>
            <h3
              className="small-movie-card__title"
            >
              <span
                className="small-movie-card__link"
              >
                Interstate 60: Episodes of the Road
              </span>
            </h3>
          </a>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
  ",",
]
`;
